src/pykeen/nn/representation.py

```diff
@@ -1398,3 +1399,94 @@ def __init__(
         assignment[mask, 1] = torch.arange(backfill.max_id)
 
         super().__init__(assignment=assignment, bases=[base, backfill], **kwargs)
+
+
+class TransformedRepresentation(Representation):
+    """
+    A (learnable) transformation upon base representations.
+
+    In the following example, we create representations which are obtained from a trainable transformation of fixed
+    random walk encoding features. We first load the dataset, here Nations:
+
+    >>> from pykeen.datasets import get_dataset
+    >>> dataset = get_dataset(dataset="nations")
+
+    Next, we create a random-walk positional encoding of dimension 32:
+
+    >>> from pykeen.nn import init
+    >>> dim = 32
+    >>> initializer = RandomWalkPositionalEncoding(triples_factory=dataset.training)
+
+    For the transformation, we use a simple 2-layer MLP
+
+    >>> from torch import nn
+    >>> hidden = 64
+    >>> mlp = nn.Sequential(
+    ...     nn.Linear(in_features=dim, out_features=hidden),
+    ...     nn.ReLU(),
+    ...     nn.Linear(in_features=hidden, out_features=dim),
+    ... )
+
+    Finally, the transformed representation is given as
+
+    >>> from pykeen.nn import TransformedRepresentation
+    >>> r = TransformedRepresentation(
+    ...     transformation=mlp,
+    ...     base_kwargs=dict(max_id=dataset.num_entities, shape=(dim), initializer=initializer, trainable=False),
```

Comment:
Is there a missing comma in the shape tuple

Reply:
good catch! https://github.com/pykeen/pykeen/pull/984/commits/c05584834235e57f470b59e72c8728b794a47f7b